### DISCUSSION TITLE
Migrate from premake4 - how to debug?

### DISCUSSION THREAD

**Discussion by furai** [premake/premake-core#1725] [Q&A]
Hey all,

I'm having issues with lua script that used to work with premake 4. I've fixed all the warnings it was complaining about in the console, but for some reason it's not generating the project files for me. I'm trying to target vs2019.

Any guides on how to debug it better to see why it's silently failing?

Cheers,
Furai

**Comment by nickclark2016** [comment]
Can you show us your premake file?

**Reply by furai** [reply]
I'd first have to clean it up. I'm mostly after some debugging options.

**Comment by starkos** [comment]
There is an [old article on debugging here](https://premake.github.io/docs/Debugging-Scripts/), not sure if it still works? TBH, I pretty much always litter a bunch of `print()`-s around the code.
